improvement/ZENKO-647-backbeat_API_Routes_Doc

wabernat opened the pull request:
Changed routes to backbeat paths for metrics.md. Copy edited as well.

wabernat committed:
Update metrics.md

wabernat committed:
Zenko-647: Backbeat API changes

Changed Backbeat API route info. Light copyedit.

wabernat committed:
Zenko-647

One more change (added a return line)

jonathan-gramain commented on docs/metrics.md:
```diff
@@ -26,37 +26,36 @@ HTTP code.
-Routes are organized in the following fashion:
-`/_/metrics/<extension-type>/<location-name/[<metric-type>]/[<bucket>]/[<key>]?[versionId=<version-id>]`
+Routes are organized as follows:
+`/_/backbeat/api/<extension-type>/<location-name/[<metric-type>]/[<bucket>]/[<key>]?[versionId=<version-id>]`
```
We have to keep the `/metrics` after `api`

wabernat replied:
Thanks!

jonathan-gramain commented on docs/metrics.md:
```diff
@@ -26,37 +26,36 @@ HTTP code.
+   object, it contains the object's version ID.
 
-### `/_/metrics/crr/<location-name>`
+### `/_/backbeat/api/<location-name>`
```
Add `/metrics/crr` between `api` and `/<location-name>` (same for the following occurrences).

jonathan-gramain commented:
@wabernatScality can you change the branch name to e.g. `improvement/ZENKO-647-backbeatAPIRoutesDoc`? It has to contain the ticket name to have an automatic link from Jira ticket, and the `improvement/` prefix is there to allow bert-e to merge it as it is one of its recognized prefixes (AFAIK `feature/`, `bugfix/` and `improvement/` can be used).

wabernat committed:
Zenko-647: CRR pause & resume

Changed routes. Torqued down the text. Changed future-tense design spec to present-tense "designed" description.

wabernat committed:
Fixed new backbeat routes per Jonathan's request

added "metrics/crr/" to all changed paths.

wabernat committed:
fixed one more instance 

...missed a spot!

wabernat commented:
Complied (I hope!) with Jonathan's requests.

bennettbuchanan commented on docs/metrics.md:
```diff
@@ -26,37 +26,36 @@ HTTP code.
-Routes are organized in the following fashion:
-`/_/metrics/<extension-type>/<location-name/[<metric-type>]/[<bucket>]/[<key>]?[versionId=<version-id>]`
+Routes are organized as follows:
+`/_/backbeat/api/metrics/crr/<extension-type>/<location-name/[<metric-type>]/[<bucket>]/[<key>]?[versionId=<version-id>]`
```
'crr' is an `<extension-type>` so we don't want to add this here while also keeping the `<extension-type>` placeholder. Alternatively, we could remove `/<extension-type>/` since 'crr' is only supported in the API (but may be extended later). Either way is good for me 😸

wabernat replied:
Let's not close any doors! <extension-type> it is.

wabernat replied:
By which I meant to say: "extension-type it is." or somesuch...

bennettbuchanan reviewed (changes requested):
Nice edits! Just one comment about a change in the route.
